File: 04_analyze/Covid/code/analyze.py

# import pandas, numpy, and matplotlib
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

# data load 
covid = pd.read_csv("03_build/weekSIR/output/cases_timeseries_plot.csv")


# data clean
covid['week'] = pd.to_datetime(covid['week'])

Neighbor = covid[covid.treat == "Neighboring_prefs"]
Neighbor = Neighbor.set_index('week')
Yama = covid[covid.treat == "Yamanashi"]
Yama = Yama.set_index('week')


# plot
fig, axes = plt.subplots()

plt.xticks(rotation=45)
axes.plot(Neighbor.index, Neighbor['newcase_day_per'], color = "#ff9900")
axes.plot(Yama.index, Yama['newcase_day_per'], color = "#339900")
axes.axvline(pd.to_datetime("2020-04-07"), color='gray', linestyle='dashed', lw=2)
axes.text(pd.Timestamp("2020-04-07"), 28, "The first \n state of emergency", color = 'gray')
axes.axvline(pd.to_datetime("2020-07-22"), color='gray', linestyle='dashed', lw=2)
axes.text(pd.Timestamp("2020-07-22"), 23, "Go To Campaign", color = 'gray')
axes.axvline(pd.to_datetime("2021-01-07"), color='gray', linestyle='dashed', lw=2)
axes.text(pd.Timestamp("2021-01-07"), 28, "The second \n state of emergency", color = 'gray')
plt.legend(['Neighboring 5 prefectures', "Yamanashi"], loc = "center left")
plt.xlabel("Week")
plt.ylabel("New infection cases per 100,000 population")
plt.title("Time series of new infection cases by week")
plt.ylim(-2.5, 35)
plt.savefig("04_analyze/Covid/output/covid_plot.png", format="png", dpi=250, pad_inches=0.1, bbox_inches="tight")

File: 04_analyze/Covid_cases_plot/code/analyze.R
library(tidyverse)

# data load ------
wPrefCovid <- read_csv(here::here("03_build/weekSIR/output/cases_timeseries_plot.csv"))

# New infection cases in time series----------

COVID_caseplot <- ggplot(data = wPrefCovid,
                         mapping = aes(x = week,
                                       y = newcase_day_per,
                                       color = treat)) +
  geom_point(size = 0.1) +
  geom_line(size = 0.5)  +
  ggtitle("Time series of new infections by week") +
  xlab("Week") +
  ylab("New infections per 100000 people") +
  scale_x_date(date_labels = "%Y-%m") +
  geom_vline(xintercept = as.Date("2020-04-07"),
             linetype = "dotdash",
             color = "grey60") +
  annotate("text",
           x = as.Date("2020-04-07"),
           y = 27.5,
           label = "The first \n state of emergency",
           size = 4)+
  geom_vline(xintercept = as.Date("2020-07-22"),
             linetype = "dotdash",
             color = "grey60")+
  annotate("text",
           x = as.Date("2020-07-22"),
           y = 27.5,
           label = "Go To Campaign",
           size = 4)+
  geom_vline(xintercept = as.Date("2021-01-07"),
             linetype = "dotdash",
             color = "grey60")+
  annotate("text",
           x = as.Date("2021-01-07"),
           y = 27.5,
           label = "The second \n state of emergency",
           size = 4)+
  scale_color_manual(name = " ",
                     values = c("#339900","#ff9900"),
                     labels = c("Yamanashi", "Neighboring \n Prefectures")) +
  theme(title = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        axis.title.x = element_text(size = 16),
        axis.text.y = element_text(size = 12),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.key.size = unit(1, "cm"))

ggsave("04_analyze/Covid_cases_plot/output/Cases_by_week.png", COVID_caseplot, width = 10, height = 8, dpi = 300)
